app/controllers/application_controller.rb

```diff
@@ -6,14 +6,14 @@ class ApplicationController < ActionController::Base
   # run these methods on page load
   before_filter :current_user
 
-  private
- 
-  # Finds the User with the ID stored in the session with the key
-  # :current_user_id This is a common way to handle user login in
-  # a Rails application; logging in sets the session value and
-  # logging out removes it.
+  helper_method :current_user
+
   def current_user
-    session[:id] ||= request.remote_ip.hash + rand(1000)
-    @_current_user = session[:id]
+    @current_user ||= unless @current_user 
```

Comment:
It is not.  I like htat you both call this out though.  Because it is something ruby should do better.

But this is the crux.  There are multiple lines of logic that could be run here.

Multiple lines of logic that I want executed only 1 time.

In the original code, why are we wasting time re-setting ```@_current_user = session[:id]``` every single time the method is called?

With my code, if @current_user is defined, it returns that with 1 if-nil check only.
Then the unless lets me have complex logic that could span multiple lines to determine what the final value will be.

Ideally this syntax would be allowed in ruby (but isnt i believe) by writing a block to be executed:

```
def some_var
  @some_var_i_define_once ||= {
      session[:id] ||= some-value 
  }
end
```

But since that formatting is invalid, you need a if/unless to open a block who's result is assigned.  But, only if @some_var_i_define_once is nil.

Compare that to:

```
def some_var
  @some_var_i_define_once = if session[:id]
                                                      session[:id] 
                                                   else
                                                     session[:id] = some-value
                                                   end
end
```

This is maybe a little clearer without the weird seemingly duplicate check, but more expensive execution wise.